tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -14,6 +14,7 @@
 import time
```

Comment:
> rptest: Update timeout calculation strategy

Can you provide in the commit message some discussion about the relevance of this PR to ARM?

Reply:
For some reason, timeouts on ARM instances get longer and normal pre-calculated values getting flaky. And in order not to put really long ones, like 1800 by default, use of calculations and proper logging on why Timeout happened is a way to go. Root cause here is that throughput on these instances are lower that on x86 ones.

Reply:
Also, this is an original PR that had multiple fixes. Some commits moved here: #14808